src/runtime_src/hip/core/event.cpp

```diff
@@ -242,6 +242,28 @@ bool memory_pool_command::wait()
   return true;
 }
 
+bool mem_prefetch_command::submit()
+{
+  auto hip_mem_and_off = memory_database::instance().get_hip_mem_from_addr(m_dev_ptr);
+  auto hip_mem = hip_mem_and_off.first;
+  size_t hip_mem_off = hip_mem_and_off.second;
+  throw_invalid_value_if(!hip_mem, "Invalid prefetch buf address.");
+  throw_invalid_value_if((hip_mem->get_size() < (hip_mem_off + m_size)),
+                         "Invalid prefetch buf address or size.");
+
+  // The under xrt::bo::sync() behaves the same for both TO_DEVICE or FROM_DEVICE direction.
+  // we pick xclBOSyncDirection::XCL_BO_SYNC_BO_TO_DEVICE as input argument here.
+  hip_mem->sync(xclBOSyncDirection::XCL_BO_SYNC_BO_TO_DEVICE, m_size, hip_mem_off);
```

Comment:
We need to call async (call std::async)  function & wait for this thread to complete in wait() API.
Otherwise , this becomes a blocking call.